ui2/src/pages/listing/Listings.tsx

```diff
@@ -18,49 +25,88 @@ export const ListingsTable: React.FC<Props> = ({ services, listings }) => {
   const party = useParty();
   const { getName } = usePartyName(party);
   const ledger = useLedger();
+  const history = useHistory();
 
   const service = services.find(s => s.payload.customer === party);
 
+  const { contracts: manualFVRequests, loading: manualFVRequestsLoading } = useStreamQueries(
+    ManualFairValueCalculation
+  );
+
+  const { contracts: fairValueContracts, loading: fairValuesLoading } = useStreamQueries(FairValue);
+
   const requestDisableDelisting = async (c: CreateEvent<Listing>) => {
     if (!service) return; // TODO: Display error
     await ledger.exercise(Service.RequestDisableListing, service.contractId, {
       listingCid: c.contractId,
     });
   };
 
+  const getMarketType = (c: CreateEvent<Listing>) => {
+    const listingType = c.payload.listingType;
+    if (listingType.tag === 'Collateralized') {
+      return 'Collateralized';
+    } else {
+      return getName(listingType.value.clearinghouse);
+    }
+  };
+
   return (
-    <StripedTable
-      headings={[
-        'Provider',
-        'Client',
-        'Listing ID',
-        'Calendar ID',
-        'Traded Asset',
-        'Traded Asset Precision',
-        'Quoted Asset',
-        'Quoted Asset Precision',
-        'Action',
-      ]}
-      rows={listings.map(c => {
-        return {
-          elements: [
-            getName(c.payload.provider),
-            getName(c.payload.customer),
-            c.payload.listingId,
-            c.payload.calendarId,
-            c.payload.tradedAssetId.label,
-            c.payload.tradedAssetPrecision,
-            c.payload.quotedAssetId.label,
-            c.payload.quotedAssetPrecision,
-            party === c.payload.customer && (
-              <Button floated="right" className="ghost" onClick={() => requestDisableDelisting(c)}>
-                Disable
-              </Button>
-            ),
-          ],
-        };
-      })}
-    />
+    <>
+      <StripedTable
+        headings={[
+          'Provider',
+          'Client',
+          'Cleared By',
+          'Listing ID',
+          'Calendar ID',
+          'Traded Asset',
+          'Traded Asset Precision',
+          'Quoted Asset',
+          'Quoted Asset Precision',
+          'Fair Value',
+          'Action',
+        ]}
+        rowsClickable
```

Comment:
Sadly I dont think this will work - the row being clickable overlaps with the button in the row. Making the click functionality for the button possibly not work.. its pretty poor design on my part... in order to have clickable rows and buttons there needs to be some kind of click listener added. So the DOM knows if the user clicked the row, or the button in the row.

Reply:
basically `rowsClickable` only works for tables with no buttons in them

Reply:
ah OK - I can move that button to the listing's page and look into adding a listener like that (in another PR).